### PR TITLE
feat(TDI-39458): Prepare definition part of MarkLogic on TCOMP

### PR DESCRIPTION
**Pull request by dmytro-sylaiev**
* Also write few unit tests

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1204/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1204/)

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicbulkload/MarkLogicBulkLoadDefinition.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package org.talend.components.marklogic.tmarklogicbulkload;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicbulkload/MarkLogicBulkLoadDefinition.java]: Copyright

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicbulkload/MarkLogicBulkLoadProperties.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package org.talend.components.marklogic.tmarklogicbulkload;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicbulkload/MarkLogicBulkLoadProperties.java]: Copyright

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicCloseDefinition.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
+package org.talend.components.marklogic.tmarklogicclose;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicCloseDefinition.java]: copyright

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionDefinition.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
+package org.talend.components.marklogic.tmarklogicconnection;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionDefinition.java]: copyright

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionProperties.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
+package org.talend.components.marklogic.tmarklogicconnection;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionProperties.java]: Copyright

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionProperties.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+    public StringProperty host = PropertyFactory.newString("host");
+
+    public Property<Integer> port = PropertyFactory.newInteger("port", 8000);
+
+    public StringProperty database = PropertyFactory.newString("database");
+
+    public Property<String> username = newProperty("username");
+
+    public Property<String> password = newProperty("password")
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionProperties.java]: You should use defined `UserPasswordProperties` in components-common.

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicCloseProperties.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
+package org.talend.components.marklogic.tmarklogicclose;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicCloseProperties.java]: Copyright

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionProperties.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+        }
+    }
+
+    public void afterReferencedComponent() {
+        refreshLayout(getForm(Form.MAIN));
+        refreshLayout(getForm(Form.REFERENCE));
+    }
+
+    public MarkLogicConnectionProperties(String name) {
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionProperties.java]: Not so good place for the cintructor. Could you please move it to the top.

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicoutput/MarkLogicOutputDefinition.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
+package org.talend.components.marklogic.tmarklogicoutput;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicoutput/MarkLogicOutputDefinition.java]: Copyright

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicoutput/MarkLogicOutputProperties.java [hunk 10]
```diff
@@ -0,0 +1,107 @@
+package org.talend.components.marklogic.tmarklogicoutput;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicoutput/MarkLogicOutputProperties.java]: copyright

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicoutput/MarkLogicOutputProperties.java [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+        fields.add(docIdField);
+        fields.add(docContentField);
+        Schema initialSchema = Schema.createRecord("marklogic", null, null, false, fields);
+        initialSchema.addProp(TALEND_IS_LOCKED, "true");
+        fields.clear();
+        schema.schema.setValue(initialSchema);
+
+        //TODO refactor it
+        schemaFlow.schema.setValue(initialSchema);
+//    errMessage
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicoutput/MarkLogicOutputProperties.java]: formatting.

File: components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/MarkLogicFamilyDefinitionTest.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+
+    private MarkLogicFamilyDefinition familyDefinition = null;
+
+    @Before
+    public void init() {
+        familyDefinition = new MarkLogicFamilyDefinition();
+    }
+
+    @Test
+    public void testAllComponentsDefinitionsCreated() {
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/MarkLogicFamilyDefinitionTest.java]: To simplify this, you can use Matchers for comparing Arrays.

File: components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicCloseDefinitionTest.java [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+    @Before
+    public void init() {
+        closeDefinition = new MarkLogicCloseDefinition();
+    }
+
+    @Test
+    public void testGetFamilies() {
+        String[] expectedFamilies = new String[] {"Databases/MarkLogic", "Big Data/MarkLogic"};
+
+        String[] actualFamilies = closeDefinition.getFamilies();
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicCloseDefinitionTest.java]: Next tests are using different scenario, you didn't create local variable for them. You should choose 1 strategy and make them similar.

File: components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicCloseDefinitionTest.java [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+    @Test
+    public void testIsStartable() {
+        assertTrue(closeDefinition.isStartable());
+    }
+
+    @Test
+    public void testGetSupportedConnectorTopologies() {
+        Set<ConnectorTopology> connectorTopologies = closeDefinition.getSupportedConnectorTopologies();
+
+        assertThat(connectorTopologies, contains(ConnectorTopology.NONE));
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicCloseDefinitionTest.java]: You can use 1 assert with smth like contains only.

File: components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicbulkload/MarkLogicBulkLoadDefinitionTest.java [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+    @Before
+    public void init() {
+        bulkLoadDefinition = new MarkLogicBulkLoadDefinition();
+    }
+
+    @Test
+    public void testGetFamilies() {
+        String[] expectedFamilies = new String[] {"Databases/MarkLogic", "Big Data/MarkLogic"};
+
+        String[] actualFamilies = bulkLoadDefinition.getFamilies();
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicbulkload/MarkLogicBulkLoadDefinitionTest.java]: The same 1 strategy without local variable.

File: components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicClosePropertiesTest.java [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+        assertEquals(1, closeProperties.getForms().size());
+        assertNotNull(closeProperties.getForm(Form.MAIN).getWidget(closeProperties.referencedComponent.getName()));
+    }
+
+    @Test
+    public void testGetReferencedComponentId() {
+        String expectedStringValue;
+        String referencedComponentId;
+
+        expectedStringValue = "SomeStringValue";
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicClosePropertiesTest.java]: you can initialize your local variable while defining it.

File: components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicClosePropertiesTest.java [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+        assertEquals(0, closeProperties.getForms().size());
+        closeProperties.setupLayout();
+        assertEquals(1, closeProperties.getForms().size());
+        assertNotNull(closeProperties.getForm(Form.MAIN).getWidget(closeProperties.referencedComponent.getName()));
+    }
+
+    @Test
+    public void testGetReferencedComponentId() {
+        String expectedStringValue;
+        String referencedComponentId;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicClosePropertiesTest.java]: This definition may be moved to actual use of variable (when initializing it with a value)

File: components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionDefinitionTest.java [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+    @Test
+    public void testComponentShouldBeStartable() {
+        assertTrue(definition.isStartable());
+    }
+
+    @Test
+    public void testGetFamilies() {
+        String[] expectedFamilies = new String[] {"Databases/MarkLogic", "Big Data/MarkLogic"};
+
+        String[] actualFamilies = definition.getFamilies();
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionDefinitionTest.java]: 1 strategy for local variable.

File: components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicinput/MarkLogicInputPropertiesTest.java [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+import org.talend.daikon.properties.presentation.Widget;
+
+public class MarkLogicInputPropertiesTest {
+
+	/**
+	 * Checks forms are filled with required widgets
+	 */
+	@Test
+	@Ignore
+	public void testSetupLayout() {
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicinput/MarkLogicInputPropertiesTest.java]: Check your formatting

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/connection/TMarkLogicCloseStandalone.java [hunk 20]
```diff
@@ -0,0 +1,19 @@
+package org.talend.components.marklogic.connection;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/connection/TMarkLogicCloseStandalone.java]: copyright

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/connection/TMarkLogicConnectionStandalone.java [hunk 21]
```diff
@@ -0,0 +1,20 @@
+package org.talend.components.marklogic.connection;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/connection/TMarkLogicConnectionStandalone.java]: Copyright

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicBulkLoad.java [hunk 22]
```diff
@@ -0,0 +1,19 @@
+package org.talend.components.marklogic.runtime;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicBulkLoad.java]: Copyright

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicReader.java [hunk 23]
```diff
@@ -0,0 +1,52 @@
+package org.talend.components.marklogic.runtime;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicReader.java]: Copyright

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSink.java [hunk 24]
```diff
@@ -0,0 +1,40 @@
+package org.talend.components.marklogic.runtime;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSink.java]: copyright

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSink.java [hunk 25]
```diff
@@ -0,0 +1,40 @@
...
+        return null;
+    }
+
+    @Override
+    public ValidationResult validate(RuntimeContainer container) {
+        return null;
+    }
+
+    @Override
+    public ValidationResult initialize(RuntimeContainer container, ComponentProperties properties) {
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSink.java]: `Validation.OK` ?

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSource.java [hunk 26]
```diff
@@ -0,0 +1,44 @@
+package org.talend.components.marklogic.runtime;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSource.java]: Copyright

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSink.java [hunk 27]
```diff
@@ -0,0 +1,40 @@
...
+import org.talend.components.api.component.runtime.WriteOperation;
+import org.talend.components.api.container.RuntimeContainer;
+import org.talend.components.api.properties.ComponentProperties;
+import org.talend.daikon.NamedThing;
+import org.talend.daikon.properties.ValidationResult;
+
+import java.io.IOException;
+import java.util.List;
+
+public class MarkLogicSink implements Sink {
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSink.java]: Should this class be a child of **MarkLogicSourceOrSink** ?

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicWriteOperation.java [hunk 28]
```diff
@@ -0,0 +1,32 @@
...
+
+public class MarkLogicWriteOperation implements WriteOperation<IndexedRecord> {
+
+    @Override
+    public void initialize(RuntimeContainer adaptor) {
+
+    }
+
+    @Override
+    public Map<String, Object> finalize(Iterable<IndexedRecord> writerResults, RuntimeContainer adaptor) {
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicWriteOperation.java]: `Result.toMap()` ?

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSourceOrSink.java [hunk 29]
```diff
@@ -0,0 +1,53 @@
+package org.talend.components.marklogic.runtime;
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSourceOrSink.java]: Copyright

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSink.java [hunk 30]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Override
+    public WriteOperation<?> createWriteOperation() {
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicSink.java]: Create defined write operation. You have it!

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/connection/TMarkLogicCloseStandalone.java [hunk 31]
```diff
@@ -0,0 +1,19 @@
...
+import org.talend.components.api.component.runtime.ComponentDriverInitialization;
+import org.talend.components.api.container.RuntimeContainer;
+import org.talend.daikon.properties.Properties;
+import org.talend.daikon.properties.ValidationResult;
+
+public class TMarkLogicCloseStandalone implements ComponentDriverInitialization {
+
+    @Override
+    public void runAtDriver(RuntimeContainer container) {
+        System.out.println("Closed");
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/connection/TMarkLogicCloseStandalone.java]: I think it's not closed properly )) `System.out.close();`

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicBulkLoad.java [hunk 32]
```diff
@@ -0,0 +1,19 @@
...
+import org.talend.components.api.component.runtime.ComponentDriverInitialization;
+import org.talend.components.api.container.RuntimeContainer;
+import org.talend.daikon.properties.Properties;
+import org.talend.daikon.properties.ValidationResult;
+
+public class MarkLogicBulkLoad implements ComponentDriverInitialization {
+
+    @Override
+    public void runAtDriver(RuntimeContainer container) {
+
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicBulkLoad.java]: `System.out.println("Loaded")` ?

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicWriteOperation.java [hunk 33]
```diff
@@ -0,0 +1,32 @@
...
+        return null;
+    }
+
+    @Override
+    public Writer<IndexedRecord> createWriter(RuntimeContainer adaptor) {
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicWriteOperation.java]: You have **Writer** I see it below!

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicWriter.java [hunk 34]
```diff
@@ -0,0 +1,30 @@
+package org.talend.components.marklogic.runtime;
+
+import org.apache.avro.generic.IndexedRecord;
+import org.talend.components.api.component.runtime.WriteOperation;
+import org.talend.components.api.component.runtime.Writer;
+
+import java.io.IOException;
+
+public class MarkLogicWriter implements Writer<IndexedRecord> {
```
Comment (mbasiuk-talend) [inline components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicWriter.java]: Need to discuss if we should use `Writer<IndexedRecord>` or `Writer<Result>`.

File: components/components-marklogic/pom.xml [hunk 35]
```diff
@@ -0,0 +1,76 @@
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (mbasiuk-talend) [inline components/components-marklogic/pom.xml]: One small remark, as we have similar split in salesforce, we need to think about moving Integration tests to sub module.

File: components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicWriteOperation.java (hunk 33 above)
Comment (dmytro-sylaiev) [reply components/components-marklogic/components-marklogic-runtime/src/main/java/org/talend/components/marklogic/runtime/MarkLogicWriteOperation.java]: Runtime part methods are only auto-generated stubs now

**Comment by dmytro-sylaiev** [comment]
Delete all comments about copyrights (they was added in every class)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1208/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1208/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1210/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1210/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1212/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1212/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Adsylaiev%2FTDI-39458_prepare_definition_part_for_marklogic)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1214/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1214/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Adsylaiev%2FTDI-39458_prepare_definition_part_for_marklogic)

File: components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionProperties.java (hunk 6 above)
Comment (dmytro-sylaiev) [reply components/components-marklogic/components-marklogic-definition/src/main/java/org/talend/components/marklogic/tmarklogicconnection/MarkLogicConnectionProperties.java]: `UserPasswordProperties` were changed and stores not usefull properties, so I would never use any of methods inside this class, only 2 fields. 
Not very useful here

File: components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicCloseDefinitionTest.java (hunk 14 above)
Comment (dmytro-sylaiev) [reply components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/tmarklogicclose/MarkLogicCloseDefinitionTest.java]: Not sure it is so important for writing specific matcher (I know no one with this behavior)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1216/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1216/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Adsylaiev%2FTDI-39458_prepare_definition_part_for_marklogic)

File: components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/MarkLogicFamilyDefinitionTest.java [hunk 36]
```diff
@@ -0,0 +1,66 @@
...
+        List<Class> expectedDefinitions = new ArrayList<>();
+        expectedDefinitions.add(MarkLogicInputDefinition.class);
+        expectedDefinitions.add(MarkLogicOutputDefinition.class);
+        expectedDefinitions.add(MarkLogicConnectionDefinition.class);
+        expectedDefinitions.add(MarkLogicCloseDefinition.class);
+        expectedDefinitions.add(MarkLogicBulkLoadDefinition.class);
```
Comment (gonchar-ivan) [inline components/components-marklogic/components-marklogic-definition/src/test/java/org/talend/components/marklogic/MarkLogicFamilyDefinitionTest.java]: Arrays.asList()

**Pull request merged**